### PR TITLE
feat: Adds support for spaces & adds null support for regex flag

### LINKED ISSUE #103: flag property is null for text elements with validation regex
### Brief bug description

When fetching elements from Kontent, the flag property is set to null even though the type expects string or undefined for text elements with validation regex.

This causes the generated model to throw typescript errors.

### Repro steps

1. Create a Content Model that mimics the screenshot below
2. Hook a NodeJS app into your Kontent project
3. Create a new ManagementClient object (e.g. managementClient)
4. Call managementClient.listContentTypes().toAllPromise()
5. The returning element will have null set for the flags property

### Expected behavior

The flag property should be set to either a string or undefined.

### Test environment

 - Platform/OS: JS SDK
 - Version 4.0.0

### Screenshots
Text Element created
image

When fetched from Kontent
image

Interface
image

Typescript Error
image

### PR DESCRIPTION
**Pull request by Nickm615**
### Motivation

Which issue does this fix? Fixes #103 

### Checklist

- [✔ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [✔ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [✔ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

I just edited the text element data interface to accept text or null for the flag property, since the property usually is returned null from the API.

### CODE REVIEW

File: lib/models/spaces/space-models.ts [hunk 1]
```diff
@@ -14,20 +14,23 @@ export namespace SpaceModels {
...
         codename?: string;
+        webSpotlightRootItem: SharedModels.ReferenceObject
     }
 
     export class Space implements SharedModels.IBaseModel<SpaceContracts.ISpaceContract> {
 
         public id!: string;
         public name!: string;
         public codename!: string;
+        webSpotlightRootItem!: SharedModels.ReferenceObject
```
Comment (Enngage) [inline lib/models/spaces/space-models.ts:25]: Shouldn't there be a 'public' identifier?

File: lib/contracts/space-contracts.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
+import { SharedContracts } from "./shared-contracts";
+
 export namespace SpaceContracts {
     export interface ISpaceContract {
         id: string;
         name: string;
         codename: string;
+        webSpotlightRootItem: SharedContracts.IReferenceObjectContract
```
Comment (Enngage) [inline lib/contracts/space-contracts.ts:8]: Is this property really required or can it be optional?

**Review by Enngage — CHANGES_REQUESTED** [review]
@Nickm615 can you please have a look at my 2 comments?

File: lib/models/spaces/space-models.ts (hunk 1 above)
Comment (Nickm615) [reply lib/models/spaces/space-models.ts:25]: I didn't intend to include these changes in these branch, my mistake. I'll fix this when I submit the root item update later this week.

Comment (Nickm615) [reply lib/models/spaces/space-models.ts]: Fixed

File: lib/models/spaces/space-models.ts [hunk 3]
```diff
@@ -14,20 +14,23 @@ export namespace SpaceModels {
     export interface IAddSpaceData {
         name: string;
         codename?: string;
+        webSpotlightRootItem?: SharedModels.ReferenceObject
```
Comment (Enngage) [inline lib/models/spaces/space-models.ts:17]: I believe this should also use the `web_spotlight_root_item` but please verify

**Review by Enngage — APPROVED** [review]

**Pull request merged**